runtime/near-vm-runner/src/runner.rs

```diff
@@ -108,3 +108,53 @@ impl VMKind {
         }
     }
 }
+
+/// Type returned by any VM instance after loading a contract and executing a
+/// function inside.
+#[derive(Debug, PartialEq)]
+pub enum VMResult {
+    /// Execution was not able to start because preconditions were not met.
+    NotRun(VMError),
+    /// Execution started but hit an error.
+    Aborted(VMOutcome, VMError),
+    /// Execution finished without error.
+    Ok(VMOutcome),
+}
+
+impl VMResult {
+    pub fn not_run(error: VMError) -> VMResult {
+        VMResult::NotRun(error)
+    }
+    #[cfg_attr(not(feature = "protocol_feature_function_call_weight"), allow(unused_mut))]
+    pub fn abort(mut logic: VMLogic, error: VMError) -> VMResult {
+        let outcome = logic.compute_outcome_and_distribute_gas();
+        VMResult::aborted(outcome, error)
+    }
+    pub fn aborted(outcome: VMOutcome, error: VMError) -> VMResult {
+        VMResult::Aborted(outcome, error)
+    }
+    pub fn ok(outcome: VMOutcome) -> VMResult {
+        VMResult::Ok(outcome)
+    }
```

Comment:
Similarly to above, I think this constructor-method provide a fair bit of value if it took `logic` and called the `compute_outcome_and_distribute_gas` for you. As-is it only serves to provide multiple ways to construct the same variant.

Reply:
I guess a part of an intent here is to make it possible to modify the ADT definition itself without changing the callers? I guess could see that being valuable if we anticipate further experimentation in this area.

Reply:
Yes, this was the intent here. Or actually, I came from the other side, where the `enum` was a `struct` before.
That said, I think if we all agree on this being an enum, then it makes sense to just remove the constructors.

Reply:
An alternative approach would be to wrap the enum into a newtype so that the variants aren't directly exposed (thus forcing use of the methods to construct and inspect the state here), though that sounds like it might be too heavy handed.

Reply:
I considered this alternative after you mentioned it. But I agree, it's too clunky and not really necessary IMO.